tunneldns/tunnel.go

```diff
@@ -68,6 +68,12 @@ func Command(hidden bool) *cli.Command {
 				Value:   cli.NewStringSlice("https://162.159.36.1/dns-query", "https://162.159.46.1/dns-query", "https://[2606:4700:4700::1111]/dns-query", "https://[2606:4700:4700::1001]/dns-query"),
 				EnvVars: []string{"TUNNEL_DNS_BOOTSTRAP"},
 			},
+			&cli.IntFlag{
+				Name:    "max-upstream-conns",
+				Usage:   "Maximum concurrent connections to upstream, unlimited by default",
+				Value:   0,
```

Comment:
I think 0 is a dangerous default here, because it'll cause the bad behaviour. Let's make the default a positive number. I'm not sure what is appropriate here... 10? Maybe 100? And let's put it into a `const` so that other code can reference the default too (e.g. `func configureBootstrap` and `func MaxUpstreamConnectionsOrDefault`).